explorer/fuzzing/ast_to_proto.cpp

```diff
@@ -750,8 +750,12 @@ static auto DeclarationToProto(const Declaration& declaration)
            choice.alternatives()) {
         auto* alternative_proto = choice_proto->add_alternatives();
         alternative_proto->set_name(alternative->name());
-        *alternative_proto->mutable_signature() =
-            TupleLiteralExpressionToProto(alternative->signature());
+        if (auto sig = alternative->signature()) {
+          *alternative_proto->mutable_signature() =
+              TupleLiteralExpressionToProto(**sig);
+        } else {
+          CARBON_CHECK(!alternative_proto->has_signature());
+        }
```

Comment:
I'm a little confused about why you'd write this CARBON_CHECK; I don't think we typically CHECK !has_field on `else` where they're not set, and I don't see a need for it. That is, this doesn't feel like validation because the other path is where it would've been set.

e.g., https://github.com/carbon-language/carbon-lang/blob/trunk/explorer/fuzzing/ast_to_proto.cpp#L174 or https://github.com/carbon-language/carbon-lang/blob/trunk/explorer/fuzzing/ast_to_proto.cpp#L482 have no `else`, and look equivalent.

Reply:
Mostly I just felt it was subtle to depend on the difference between `signature` being absent and being set to default values, given that the proto API hides a lot of the difference between those states. But I've not worked a lot with proto before so I'm happy to remove this if it's not useful to you as a reader :)

Reply:
I'm good with removing it, I think it's reasonable to assume that fields are only being set in one place, and with proto the expected behavior is a default of !has_*